components/multipart-stream-builder.rst

```diff
@@ -0,0 +1,62 @@
+Multipart Stream Builder
+========================
+
+A multipart stream is a special kind of stream that is used to transfer files over HTTP. There is currently no PSR-7 support for multipart streams as they are considered to be normal streams with a special content. A multipart stream HTTP request may look like this:
+
+.. code-block:: none
+
+    POST / HTTP1/1
+    User-Agent: curl/7.21.2 (x86_64-apple-darwin)
+    Host: localhost:8080
+    Accept: */*
+    Content-Length: 1143
+    Expect: 100-continue
+    Content-Type: multipart/form-data; boundary=----------------------------83ff53821b7c
+
+    ------------------------------83ff53821b7c
+    Content-Disposition: form-data; name="img"; filename="a.png"
+    Content-Type: application/octet-stream
+
+    ?PNG
+
+    IHD?wS??iCCPICC Profilex?T?kA?6n??Zk?x?"IY?hE?6?bk
+    Y?<ߡ)??????9Nyx?+=?Y"|@5-?M?S?%?@?H8??qR>?׋??inf???O?????b??N?????~N??>?!?
+    ??V?J?p?8?da?sZHO?Ln?}&???wVQ?y?g????E??0
+     ??
+       IDAc????????-IEND?B`?
+    ------------------------------83ff53821b7c
+    Content-Disposition: form-data; name="foo"
+
+    bar
```

Comment:
can we make this example match exactly with the example below? notably show where the text/plain content type of baz shows up?